strapi/strapi
# fix(content-manager): translate content-type fields in List and Edit …

### What does it do?

- Adds ability to translate a content-types' attributes in both the list view (table headers) and edit view (form labels)
- Adds ability to translate components' attributes wherever the component is used

The translation ids are computed with:

`content-manager.{content-types | components}.{contentTypeUid | componentUid}.{attributeName}`

### Why is it needed?

- It was possible in Strapi 4

### How to test it?

In `examples/getstarted/src/admin/app.jsx` provide a config like the one below.
Then change the language of your app in your profile to one of the available locales

```js
const config = {
  locales: ['it', 'es', 'en', 'en-GB', 'fr'],
  translations: {
    fr: {
      // Keys scoped to a content-type, tranlsates the attribute for that content-type
      'content-manager.content-types.api::kitchensink.kitchensink.short_text': 'translated short text',
      'content-manager.content-types.api::kitchensink.kitchensink.single_compo':
        'translated single compo',
      // Keys scoped to a component, translates the component attribute for any content-type that uses it
      'content-manager.components.basic.relation.categories': 'translated dynamic zone compo attribute',
      'content-manager.components.default.temp.name': 'translated default.temp name',
      'content-manager.components.basic.simple.name': 'translated basic.simple name',
      'content-manager.components.basic.simple.nested': 'translated nested compo',
    },
  },
};
```
- Providing a translation for a root level attribute (text, component, dynamic zone, integer, etc...) should only translate within the scope of its content-type
- Providing a translation for a component (non-repeatable, repeatable, and dynamic zone) should translate within the scope of the component and therefore in _any_ content-type that is using it.

### Related issue(s)/PR(s)

resolves: https://github.com/strapi/strapi/issues/20461
